// Copyright 2018 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kvstore;

// A simple key-value storage service
service KVStore {
  // Provides a value for each key request
  rpc GetValue (stream GetRequest) returns (stream GetResponse) {}
  rpc SetValue (stream SetRequest) returns (stream SetResponse) {}
}

// The request message containing the key
message GetRequest {
  string key = 1;
  int32 id = 2;
}

// The response message containing the value associated with the key
message GetResponse {
  string value = 1;
  bool status = 2;
  int32 request_id = 3;
}

// The set request message containing the key and value
message SetRequest {
  string key = 1;
  string value = 2;
  int32 id = 3;
}

// The set response message containing status and request ID
message SetResponse {
  bool status = 1;
  int32 request_id = 2;
}
